### PR TITLE
Include sandboxing data in the permissions graph change request

### PR DESCRIPTION
**Pull request by alxnddr**
## Changes

This PR includes **Group Table Access Policies or GTAP** in the permissions graph request to avoid having orphaned policies. Previously, users could change a table permission to "Sandboxed", create a GTAP which did not wait the permissions graph saving and then just cancel the permissions changes. 

## How to verify

- Run Metabase EE
- Go to the [data permissions](http://localhost:3000/admin/permissions/data/) -> All Users -> Sample Database
- Change the Accounts table permission to Sandboxed, add any mapping, and click Save
- Ensure header with Save and Cancel buttons has appeared, click Save
- Refresh the page and ensure the permission is sandboxed with the correct mapping
- Open the sandboxing settings for the Accounts table and ensure they are exactly as you set them

### CODE REVIEW

**Comment by alxnddr** [comment]
@noahmoss can you please change the endpoint so that it always return the permission graph? It does not return anything if I change sandboxing data

File: enterprise/frontend/src/metabase-enterprise/sandboxes/components/AttributeMappingEditor/AttributeMappingEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      </div>
+    }
+    renderKeyInput={({ value, onChange }) => (
+      <AttributePicker
+        value={value}
+        onChange={onChange}
+        attributesOptions={(value ? [value] : []).concat(attributesOptions)}
+      />
+    )}
+    render
```
Comment (gusaiani) [inline enterprise/frontend/src/metabase-enterprise/sandboxes/components/AttributeMappingEditor/AttributeMappingEditor.tsx:54]: Is this anything? 🤔

**Review by gusaiani — APPROVED** [review]
FE overall looking good 👍

Left a minor comment inlined.

Also, atm type-check is red in CI.

**Comment by alxnddr** [comment]
@noahmoss can you please have a look at the failing BE checks?

File: enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj [hunk 2]
```diff
@@ -62,6 +62,15 @@
...
   (db/select-one GroupTableAccessPolicy :id id))
 
+(api/defendpoint POST "/validate"
+  "Validate a sandbox which may not have yet been saved. This runs the same validation that is performed when the
+  sandbox is saved, but doesn't actually save the sandbox."
+  [:as {{:keys [table_id card_id]} :body}]
+  {table_id             pos-int?
+   card_id              [:maybe pos-int?]}
+  (gtap/check-columns-match-table {:table_id table_id
+                                   :card_id  card_id}))
```
Comment (escherize) [inline enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:72]: if `:card_id` is not passed into `check-columns-match-table`, then [no check will occur](https://github.com/metabase/metabase/blob/gtap-in-permissions-graph/enterprise/backend/src/metabase_enterprise/sandbox/models/group_table_access_policy.clj#L88-L96). 

Could you add a test for it, and is that intended behavior?

**Review by escherize — CHANGES_REQUESTED** [review]
Looks good, but had a question about when card_id is null. We could require it in the `defendpoint` if it's needed

File: enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj (hunk 2 above)
Comment (alxnddr) [reply enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:72]: `card_id` is optional, it is possible to create a policy without it

Comment (escherize) [reply enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:72]: So if it's optional, will any policy without one be valid? I'm not actually sure, but a test with a nil card_id would be nice to see.

Comment (noahmoss) [reply enterprise/backend/src/metabase_enterprise/sandbox/api/gtap.clj:72]: @escherize yeah this is intended. a sandboxing policy without a `card_id` is one that instead uses user attributes to filter on a column. and the validation we're currently doing is only relevant for sandboxing that uses an underlying card to replace a table.

I'll add a test that just checks that validation passes when `card_id` is nil.

**Review by escherize — APPROVED** [review]
